src/datadog/remote_config/remote_config.cpp

```diff
@@ -173,28 +159,41 @@ std::vector<ConfigMetadata> RemoteConfigurationManager::process_response(
     if (client_configs_it == json.cend()) {
       if (!applied_config_.empty()) {
         std::for_each(applied_config_.cbegin(), applied_config_.cend(),
-                      [this, &config_updates](const auto it) {
-                        auto updated = revert_config(it.second);
-                        config_updates.insert(config_updates.end(),
-                                              updated.begin(), updated.end());
+                      [this](const auto& it) {
+                        for (const auto& listener :
+                             listeners_per_product_[it.second.product]) {
+                          listener->on_revert(it.second);
+                        }
                       });
         applied_config_.clear();
       }
-      return config_updates;
+
+      for (const auto& listener : listeners_) {
+        listener->on_post_process();
+      }
+      return;
     }
 
     // Keep track of config path received to know which ones to revert.
     std::unordered_set<std::string> visited_config;
```

Comment:
I'm not familiar with this json library, but if the string views you're using down below persist after the loop it might be worth changing this as well to use views.

Reply:
Good catch. Thank you.

Reply:
Addressed in a32272c1539e812f597285b763d8d4f5fe93df94